src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -577,6 +580,39 @@ public boolean hasTag(String tag) {
 				log.printError();
 				return null;
 			}
+			Matcher literalClassInfoMatcher = LITERAL_CLASSINFO_PARSE.matcher(expr);
+			if (literalClassInfoMatcher.matches()) {
+				String literalString = literalClassInfoMatcher.group(1);
+				String classInfoString = literalClassInfoMatcher.group(2);
+				ClassInfo<?> classInfo = Classes.parse(classInfoString, ClassInfo.class, context);
+				if (classInfo == null) {
+					log.printError();
+					return null;
+				}
+				Parser<?> classInfoParser = classInfo.getParser();
+				if (classInfoParser == null || !classInfoParser.canParse(context)) {
+					Skript.error("A '" + classInfoString  + "' cannot be parsed.");
+					log.printError();
+					return null;
+				}
+				boolean canParse = false;
+				for (ClassInfo<?> exprClassInfo : exprInfo.classes) {
+					if (exprClassInfo.getC().isAssignableFrom(classInfo.getC())) {
+						canParse = true;
+						break;
+					}
+				}
+				if (!canParse) {
+					Skript.error(expr + " " + Language.get("is") + " " + notOfType(types));
+					log.printError();
+					return null;
+				}
+				Object parsedObject = classInfoParser.parse(literalString, context);
+				if (parsedObject != null) {
+					log.printLog();
+					return new SimpleLiteral<>(parsedObject, false, new UnparsedLiteral(literalString));
```

Comment:
it'd be great if we could add a comment to this return to explain why we're doing this